streams-tests/shared/src/test/scala/zio/stream/ZSinkSpec.scala

```diff
@@ -167,12 +167,34 @@ object ZSinkSpec extends ZIOBaseSpec {
             }(equalTo(Chunk(Right(3), Left("Aie"))))
           )
         ),
-        test("dropWhile")(
-          assertZIO(
-            ZStream(1, 2, 3, 4, 5, 1, 2, 3, 4, 5)
-              .pipeThrough(ZSink.dropWhile[Int](_ < 3))
-              .runCollect
-          )(equalTo(Chunk(3, 4, 5, 1, 2, 3, 4, 5)))
+        suite("dropWhile")(
+          test("happy path")(
+            assertZIO(
+              ZStream(1, 2, 3, 4, 5, 1, 2, 3, 4, 5)
+                .pipeThrough(ZSink.dropWhile[Int](_ < 3))
+                .runCollect
+            )(equalTo(Chunk(3, 4, 5, 1, 2, 3, 4, 5)))
+          ),
+          test("correct leftovers behavior") {
+            assertZIO(
+              ZStream
+                .range(0, 20, chunkSize = 3)
+                .run(ZSink.dropWhile[Int](_ <= 10).collectLeftover)
+                .map(_._2)
+            )(equalTo(Chunk(11)))
```

Comment:
Shouldn't this be `Chunk(11, 12, ..., 19)`?

Reply:
no, that exactly the bug.
once a sink succeeds it's not supposed to read anymore from upstream, due to chunking it may have leftovers it must write to downstream. in this case upstream emits chunks of 3, so when the sink completes it has a single leftover element (11) which is exactly what the assertion tests